plugin/superupdate.vim

```diff
@@ -1,45 +1,88 @@
 " superupdate.vim
 
-if exists("g:loaded_superupdate")
+if exists("g:superupdate#included")
     finish
 endif
-let g:loaded_superupdate = 1
+let g:superupdate#included = 1
+
+runtime ./vim-logger/logger.vim
+
+" ====================================================================== const =
+let s:INTEGER = 0
+let s:SHORT = 1
+let s:LONG = 2
+
+let s:SECOND = 1
+let s:MINUTE = 60 * s:SECOND
+let s:HOUR = 60 * s:MINUTE
+let s:DAY = 24 * s:HOUR
+let s:WEEK = 7 * s:DAY
+
+" ==================================================================== helpers =
+
+" ----------------------------------------- ( type = s:INTEGER ) - s:DayOfWeek -
+" obtain the day of week as an integer, three letter abbreviation or full name
+function! s:DayOfWeek ( ... )
+    if a:0 > 1
+        call WARN("DayOfWeek expects one or no parameters; found", a:0)
+    endif
+    let l:type = a:0 > 0 ? a:1 : s:INTEGER
+    if l:type == s:INTEGER
+        return strftime("%u") - 1
+    elseif l:type == s:SHORT
+        return strftime("%a")
+    elseif l:type == s:LONG
+        return strftime("%A")
+    endif
+    call WARN("DayOfWeek unknown type", l:type)
+    return -1
+endfunction
+
+" ---------------------------------------------------------- ( ) - s:Timestamp -
+" retrieve timestamp in seconds from epoch
+function! s:Timestamp ( )
+    return str2nr(strftime("%s"))
+endfunction
+
+" ==================================================================== autocmd =
 
 augroup superupdate_autocmds
     autocmd!
     "check for updates when Vim starts
-    autocmd VimEnter * call CheckForUpdate()
+    autocmd VimEnter * call g:superupdate#CheckForUpdate()
 
     "save update date if user runs an update manually
     autocmd BufDelete * if &previewwindow && &ft == "vundle" |
-                \ call SaveLastUpdate() | endif
+                \ call g:superupdate#SaveLastUpdate() | endif
 augroup END
 
-"function to save today's date as YYYYMMDD
-function! SaveLastUpdate()
-    let g:LAST_UPDATE = str2nr(strftime("%Y%m%d"))
+" ----------------------------------------- ( ) - g:superupdate#SaveLastUpdate -
+" save current timestamp
+function! g:superupdate#SaveLastUpdate()
+    call INFO("SuperUpdate: update complete")
+    let g:superupdate#last_update = s:Timestamp()
```

Comment:
Line 63 here would be an issue, currently in order to save a global variable in the viminfo file it needs to be uppercase `:help viminfo-!`. This will not be a problem when issue #4 gets resolved.